frontend/src/context/RequireAuth.tsx

```diff
@@ -2,9 +2,22 @@ import {ReactElement, useContext} from "react";
 import {AuthContext} from "./AuthProvider";
 import LoginPage from "../pages/LoginPage";
 
-export default function RequireAuth ({children} : {children:ReactElement<any,any>}){
+export default function RequireAuth({children}: { children: ReactElement<any, any> }) {
 
-    const {token} = useContext(AuthContext)
+    const {jwtDecoded} = useContext(AuthContext)
+
+    function isExpirationValid(): boolean {
+        const Now = new Date();
+        //@ts-ignore
```

Comment:
```suggestion
```

Reply:
hätte ich hier mit ts ignore stehen gelassen, da der Fall, dass jwtDecoded.exp undefined ist später im returns Block mit abgefangen wird. Wenn ich jetzt ts-ignore rausnehme bleibt der abgefangene Fall (undefined) noch als Fehler stehen?

Reply:
Ja okay, ich würde die Überprüfung lieber vorher machen, damit dein Code clean bleibt und nicht crashed falls du die Methode mal aufrufst ohne das vorher überprüft zu haben. 
```suggestion
    function isExpirationValid(): boolean {
        if (!jwtDecoded?.exp) return false

        const Now = new Date();

        const ExpirationTimeToken = new Date(jwtDecoded.exp * 1000);
        const TimeLeftToPlay: number = ExpirationTimeToken.getTime() - Now.getTime()
        return (TimeLeftToPlay > 0)
    }

    if (isExpirationValid()) {
        return children;
    } else {
        return <LoginPage/>
    }
```